tutorials/distributed_data_cache/c++11/home_automation_subscriber.cxx

```diff
@@ -0,0 +1,106 @@
+//
+// (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+//
+//  RTI grants Licensee a license to use, modify, compile, and create derivative
+//  works of the Software.  Licensee has the right to distribute object form
+//  only for use with RTI products.  The Software is provided "as is", with no
+//  warranty of any type, including any warranty for fitness for any purpose.
+//  RTI is under no obligation to maintain or support the Software.  RTI shall
+//  not be liable for any incidental or consequential damages arising out of the
+//  use or inability to use the software.
+//
+
+#include <iostream>
+#include <thread>
+
+#include "rti/rti.hpp"
+#include "rti/sub/SampleProcessor.hpp"
+#include "home_automation.hpp"
+
+void wait_for_enter(std::string query)
+{
+    std::cout << std::endl << std::endl << "Press Enter to run the query:"
+            << std::endl << "\t" << query << std::endl;
+    std::cin.get();
+}
+
+int main(int argc, char **argv)
+{
+    dds::domain::DomainParticipant participant(0);
+    dds::topic::Topic<DeviceStatus> topic(participant, "WindowStatus");
+    dds::sub::DataReader<DeviceStatus> reader(topic);
+    dds::sub::LoanedSamples<DeviceStatus> samples {};
+
+    auto print_sample = [](const DeviceStatus& sample) {
+        std::cout << sample << std::endl;
+    };
+
+    // Display all the full cache
+    wait_for_enter("reader.read()");
```

Comment:
I don't like the printting of the "Queries" before executing them, it makes the file really hard to understand. Any suggestions? @alexcamposruiz
Could we remove them and just print what we are querying, in english, and not the actual query

Reply:
Agreed